src/transformers/tokenization_utils_base.py

```diff
@@ -1742,33 +1749,44 @@ def apply_chat_template(
         # Compilation function uses a cache to avoid recompiling the same template
         compiled_template = self._compile_jinja_template(chat_template)
 
-        rendered = compiled_template.render(
-            messages=conversation, add_generation_prompt=add_generation_prompt, **self.special_tokens_map
-        )
+        if isinstance(conversation, (list, tuple)) and (
+            isinstance(conversation[0], (list, tuple)) or hasattr(conversation[0], "messages")
+        ):
+            conversations = conversation
+            batched = True
+        else:
+            batched = False
+            conversations = [conversation]
+
+        rendered = []
+        for chat in conversations:
+            if hasattr(chat, "messages"):
+                # Indicates it's a Conversation object
+                chat = chat.messages
+
+            rendered.append(
+                compiled_template.render(
+                    messages=chat, add_generation_prompt=add_generation_prompt, **self.special_tokens_map
+                )
+            )
+
+        if not batched:
+            rendered = rendered[0]
```

Comment:
should we not always return a batched output? (breaking but we can warn)

Reply:
I'm not sure - other tokenizer methods don't auto-batch a single input, right? (And sorry for taking so long to reply here!)